include/rocksdb/utilities/transaction_db.h

```diff
@@ -123,6 +126,22 @@ struct KeyLockInfo {
   bool exclusive;
 };
 
+struct DeadlockInfo {
+  TransactionID m_txn_id;
+  uint32_t m_cf_id;
+  std::string m_waiting_key;
+  bool m_exclusive;
+};
+
+struct DeadlockPath {
+  std::vector<DeadlockInfo> path;
+  bool limit_exceeded;
+  DeadlockPath(
```

Comment:
Instead of default arguments, maybe it'd be better to have two constructors for the two cases you expect (path exists or path does not exist).